crates/router/src/compatibility/stripe/payment_intents/types.rs

```diff
@@ -143,12 +144,23 @@ pub struct StripePaymentIntentRequest {
     pub client_secret: Option<pii::Secret<String>>,
     pub payment_method_options: Option<StripePaymentMethodOptions>,
     pub merchant_connector_details: Option<admin::MerchantConnectorDetailsWrap>,
+    pub mandate_id: Option<String>,
+    pub off_session: Option<bool>,
 }
 
 impl TryFrom<StripePaymentIntentRequest> for payments::PaymentsRequest {
     type Error = error_stack::Report<errors::ApiErrorResponse>;
     fn try_from(item: StripePaymentIntentRequest) -> errors::RouterResult<Self> {
-        Ok(Self {
+        let mandate_options_enum = item.payment_method_options.as_ref().and_then(|mandate| {
+            let StripePaymentMethodOptions::Card {
+                request_three_d_secure: _,
+                mandate_options,
+            }: StripePaymentMethodOptions = mandate.clone();
+            mandate_options
+        });
```

Comment:
```suggestion
        let (stripe_mandate, authentication_type) = match item.payment_method_options {
            Some(pmo) => {
                let StripePaymentMethodOptions::Card {
                    request_three_d_secure,
                    mandate_options,
                }: StripePaymentMethodOptions = pmo;
                (mandate_options, Some(request_three_d_secure.foreign_into()))
            }
            None => (None, None),
        };
```

Reply:
can we do this

`let mandate_options_enum = item.payment_method_options.as_ref().and_then(|mandate| {
            let StripePaymentMethodOptions::Card {
                request_three_d_secure: _,
                mandate_options,
            }: StripePaymentMethodOptions = mandate.to_owned();
            mandate_options
        });`

Reply:
`to_owned()` and `clone()` both are same in this context.

Reply:
ok